pkg/gui/controllers/helpers/refresh_helper.go

```diff
@@ -456,6 +458,14 @@ func (self *RefreshHelper) refreshBranches(refreshWorktrees bool) {
 		}
 	}
 
+	if !keepBranchSelectionIndex && prevSelectedBranch != nil {
+		_, idx, found := lo.FindIndexOf(self.c.Contexts().Branches.GetItems(),
```

Comment:
just confirming that GetItems() returns the filtered list, if the list is filtered?

Reply:
Yes, it calls `ListViewModel.getModel`, which ends up calling `GetFilteredList` in the case of a `FilteredListViewModel`.